lib/beaker-hostgenerator/generator.rb

```diff
@@ -104,6 +112,19 @@ def __parse_node_info_token(token)
         node_info['arbitrary_roles'] = []
       end
 
+      if node_info['host_settings']
+        # Turns a string like "{foo=bar,this=that}" into a
+        # hash like {'foo' => 'bar', 'this' => 'that'}.
+        node_info['host_settings'] = Hash[
+          node_info['host_settings'].
+          delete('{}').
+          split(',').
+          map { |keyvalue| keyvalue.split('=') }
+        ]
+      else
+        node_info['host_settings'] = {}
+      end
+
       return node_info
     end
```

Comment:
My only request as far as decomposition goes is that we mark the new functions as private, otherwise decomposition ftw!